### PR TITLE
change CORS allow origin default to allow no origins

### LINKED ISSUE #11169: Change the default for `http.cors.allow-origin`
We currently have CORS disabled by default since #7151 was merged, but once it is enabled we still allow any origin (`http.cors.allow-origin` defaults to `*`). This is a bad default as a user may just enable CORS without specifying the origins to allow, which means that any site can still issue requests to elasticsearch. We should probably default `http.cors.allow-origin` to `null` like @rashidkpc suggested in https://github.com/elastic/elasticsearch/issues/7151#issuecomment-53098415

### PR DESCRIPTION
**Pull request by jaymode**
Today, we disable CORS by default, but if a user simply enables CORS their instance of
elasticsearch will allow cross origin requests from anywhere, as the default value for allowed
origins is `*`.

This changes the default to be `null` so that no origins are allowed and the user must explicitly
specify the origins they wish to allow requests from. The documentation also mentions that there
is a security risk in using `*` as the value.

Closes #11169

### CODE REVIEW

File: docs/reference/modules/http.asciidoc [hunk 1]
```diff
@@ -55,11 +55,13 @@ Defaults to `6`.
...
 
-|`http.cors.allow-origin` |Which origins to allow. Defaults to `*`,
-i.e. any origin. If you prepend and append a `/` to the value, this will
+|`http.cors.allow-origin` |Which origins to allow. Defaults to no origins
+allowed. If you prepend and append a `/` to the value, this will
 be treated as a regular expression, allowing you to support HTTP and HTTPs.
 for example using `/https?:\/\/localhost(:[0-9]+)?/` would return the
-request header appropriately in both cases.
+request header appropriately in both cases. `*` is a valid value but is
+considered a *secruity risk* as your elasticsearch instance is open to cross origin
```
Comment (jpountz) [inline docs/reference/modules/http.asciidoc]: s/secruitysecurity/

**Comment by jpountz** [comment]
Should there be a note in the migration guide about this change?

**Comment by jaymode** [comment]
Good catch @jpountz! Added the migration docs and fixed the typo

**Comment by jpountz** [comment]
LGTM!

**Pull request merged**